metanorm/normalizers/ftp.py

```diff
@@ -0,0 +1,146 @@
+"""Normalizer for the interpretation of file name convention"""
+
+import calendar
+import logging
+import re
+import dateutil.parser
+import pythesint as pti
+from dateutil.relativedelta import relativedelta
+from dateutil.tz import tzutc
+from django.contrib.gis.geos.geometry import GEOSGeometry
+from urllib.parse import urlparse
+from .base import BaseMetadataNormalizer
+from datetime import datetime
+
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.addHandler(logging.NullHandler())
+
+
+class URLMetadataNormalizer(BaseMetadataNormalizer):
+    """ Normalizer for hardcoding information based on URLS """
+
+    def looper(self, associated_dict, raw_attributes_url, pti_function_name):
+        """looper is a helper function. It loops through the keys of "associated_dict" for finding
+        the corresponding value based on 'url' and 'pti_function_name' method of pti"""
+        for url in associated_dict.keys():
+            if raw_attributes_url.startswith(url):
+                return pti.__getattribute__(pti_function_name)(associated_dict[url])
+
+    def get_platform(self, raw_attributes):
+        """ return the corresponding platfrom based on specified ftp source """
+        if 'url' in raw_attributes:
+            urls_platforms = {
+            "ftp://ftp.remss.com/gmi": 'GPM',
+            "ftp://anon-ftp.ceda.ac.uk/neodc/esacci/sst/data/CDR_v2/Climatology/L4/v2.1":
+            'Earth Observation Satellites',
+            "ftp://ftp.gportal.jaxa.jp/standard/GCOM-W": 'GCOM-W1'}
+            return self.looper(urls_platforms, raw_attributes['url'], "get_gcmd_platform")
```

Comment:
Please fix in other methods
```suggestion
            return self.get_matching_value(urls_platforms, raw_attributes, pti.get_gcmd_platform)
```

Reply:
Done in the next commit